### PR TITLE
Improve javadoc in RollbackRuleAttribute regarding nested classes

### PR DESCRIPTION
**Pull request by chenqimiao**
In `public RollbackRuleAttribute(Class<?> clazz) `, only store the class name of the specified class, and in `private int getDepth(Class<?> exceptionClass, int depth) `, matching logic is by
`exceptionClass.getName().contains(this.exceptionName)`.
So, the enclosed class of the specified parameter of `public RollbackRuleAttribute(Class<?> clazz)` also can be match.
Suggest Improve javadoc in corresponding method.

### CODE REVIEW

**Review by sbrannen — CHANGES_REQUESTED** [review]
Please introduce a test which demonstrates that an enclosing class would result in a positive match.

**Comment by chenqimiao** [comment]
@sbrannen Thank you for your review, I have submitted a unit test to demonstrate that an enclosing class would result in a positive match.

File: spring-tx/src/main/java/org/springframework/transaction/interceptor/RollbackRuleAttribute.java [hunk 1]
```diff
@@ -54,7 +54,7 @@
 	/**
 	 * Create a new instance of the {@code RollbackRuleAttribute} class.
 	 * <p>This is the preferred way to construct a rollback rule that matches
-	 * the supplied {@link Exception} class (and subclasses).
+	 * the supplied {@link Exception} class (, subclasses and enclosed class).
```
Comment (sbrannen) [inline spring-tx/src/main/java/org/springframework/transaction/interceptor/RollbackRuleAttribute.java]:
```suggestion
	 * the supplied {@link Exception} class, its subclasses, and its nested classes.
```

**Review by sbrannen — CHANGES_REQUESTED** [review]

File: spring-tx/src/test/java/org/springframework/transaction/interceptor/RollbackRuleTests.java [hunk 2]
```diff
@@ -88,4 +88,19 @@ public void ctorArgExceptionStringNameVersionWithNull() {
...
 	}
 
+	@Test
+	public void foundEnclosedExceptionWithEnclosingException() {
+		RollbackRuleAttribute rr = new RollbackRuleAttribute(EnclosingException.class);
+		assertThat(rr.getDepth(new EnclosingException.EnclosedException())).isEqualTo(0);
+	}
+
+	@SuppressWarnings("serial")
+	static class EnclosingException extends RuntimeException{
```
Comment (sbrannen) [inline spring-tx/src/test/java/org/springframework/transaction/interceptor/RollbackRuleTests.java]:
```suggestion
	static class EnclosingException extends RuntimeException {
```

File: spring-tx/src/test/java/org/springframework/transaction/interceptor/RollbackRuleTests.java [hunk 3]
```diff
@@ -88,4 +88,19 @@ public void ctorArgExceptionStringNameVersionWithNull() {
...
+
+		@SuppressWarnings("serial")
+		static class EnclosedException extends RuntimeException{
```
Comment (sbrannen) [inline spring-tx/src/test/java/org/springframework/transaction/interceptor/RollbackRuleTests.java]:
```suggestion
		static class EnclosedException extends RuntimeException {
```

**Review by sbrannen — CHANGES_REQUESTED** [review]

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks

**Pull request merged**